fastapi/fastapi_dispatcher.py

```diff
@@ -47,7 +52,6 @@ def handle_error(self, exc):
     def _make_response(self, status_mapping, headers_tuple, content):
         self.status = status_mapping[:3]
         self.headers = dict(headers_tuple)
-        self.inner_exception = None
```

Comment:
IMO you still need to reset the inner_exception each time you process a new response

Reply:
@lmignon my understanding, but correct me if I'm wrong, is that the dispatcher will be re-initialized for each request. Now, if the assumption of the re-init per request is correct I wouldn't bother handling this differently.

Another point: even assuming you are correct, if you reset it for each iteration you might lose a previous exception. Hence, we should probably do a check if it is valued already and do not override it.

WDYT?

Reply:
@simahawk I think I was wrong :smirk: